src/test/rpc_tests.cpp

```diff
@@ -111,20 +111,20 @@ BOOST_AUTO_TEST_CASE(rpc_rawsign)
 
 BOOST_AUTO_TEST_CASE(rpc_format_monetary_values)
 {
-    BOOST_CHECK_EQUAL(write_string(ValueFromAmount(0LL), false), "0.00000000");
-    BOOST_CHECK_EQUAL(write_string(ValueFromAmount(1LL), false), "0.00000001");
-    BOOST_CHECK_EQUAL(write_string(ValueFromAmount(17622195LL), false), "0.17622195");
-    BOOST_CHECK_EQUAL(write_string(ValueFromAmount(50000000LL), false), "0.50000000");
-    BOOST_CHECK_EQUAL(write_string(ValueFromAmount(89898989LL), false), "0.89898989");
-    BOOST_CHECK_EQUAL(write_string(ValueFromAmount(100000000LL), false), "1.00000000");
-    BOOST_CHECK_EQUAL(write_string(ValueFromAmount(2099999999999990LL), false), "20999999.99999990");
-    BOOST_CHECK_EQUAL(write_string(ValueFromAmount(2099999999999999LL), false), "20999999.99999999");
+    BOOST_CHECK_EQUAL(ValueFromAmount(0LL).write(), "0.00000000");
+    BOOST_CHECK_EQUAL(ValueFromAmount(1LL).write(), "0.00000001");
+    BOOST_CHECK_EQUAL(ValueFromAmount(17622195LL).write(), "0.17622195");
+    BOOST_CHECK_EQUAL(ValueFromAmount(50000000LL).write(), "0.50000000");
+    BOOST_CHECK_EQUAL(ValueFromAmount(89898989LL).write(), "0.89898989");
+    BOOST_CHECK_EQUAL(ValueFromAmount(100000000LL).write(), "1.00000000");
+    BOOST_CHECK_EQUAL(ValueFromAmount(2099999999999990LL).write(), "20999999.99999990");
+    BOOST_CHECK_EQUAL(ValueFromAmount(2099999999999999LL).write(), "20999999.99999999");
 }
 
 static Value ValueFromString(const std::string &str)
 {
-    Value value;
-    BOOST_CHECK(read_string(str, value));
+    UniValue value;
+    BOOST_CHECK(value.setNumStr(str));
```

Comment:
Had to change this to `setNumStr` because `read()` only parses valid JSON (things like "0.00000001" is not valid JSON).